lib/cli/src/commands/publish.rs

```diff
@@ -0,0 +1,647 @@
+                        })?;
+                    builder
+                        .append_path_with_name(&normalized_path, &module.source)
+                        .map_err(|e| PublishError::ErrorBuildingPackage(module.name.clone(), e))?;
+                }
+            }
+        }
+
+        // bundle the package filesystem
+        for (_alias, path) in manifest.fs.unwrap_or_default().iter() {
+            let normalized_path = normalize_path(&cwd, path);
+            let path_metadata = normalized_path.metadata().map_err(|_| {
+                PublishError::MissingManifestFsPath(normalized_path.to_string_lossy().to_string())
+            })?;
+            if path_metadata.is_dir() {
+                builder.append_dir_all(path, &normalized_path)
+            } else {
+                return Err(PublishError::PackageFileSystemEntryMustBeDirectory(
+                    path.to_string_lossy().to_string(),
+                )
+                .into());
+            }
+            .map_err(|_| {
+                PublishError::MissingManifestFsPath(normalized_path.to_string_lossy().to_string())
+            })?;
+        }
+
+        builder.finish().ok();
+        let tar_archive_data = builder.into_inner().map_err(|_| PublishError::NoModule)?;
+        let archive_name = "package.tar.gz".to_string();
+        let archive_dir = tempfile::TempDir::new()?;
+        let archive_dir_path: &std::path::Path = archive_dir.as_ref();
+        fs::create_dir(archive_dir_path.join("wapm_package"))?;
+        let archive_path = archive_dir_path.join("wapm_package").join(&archive_name);
+        let mut compressed_archive = fs::File::create(&archive_path).unwrap();
+        let mut gz_enc = GzEncoder::new(&mut compressed_archive, Compression::best());
+
+        gz_enc.write_all(&tar_archive_data).unwrap();
+        let _compressed_archive = gz_enc.finish().unwrap();
+        let mut compressed_archive_reader = fs::File::open(&archive_path)?;
+
+        let maybe_signature_data = sign_compressed_archive(&mut compressed_archive_reader)?;
+        let archived_data_size = archive_path.metadata()?.len();
+
+        assert!(archive_path.exists());
+        assert!(archive_path.is_file());
+
+        if self.dry_run {
+            // dry run: publish is done here
+
+            println!(
+                "Successfully published package `{}@{}`",
+                package.name, package.version
+            );
+
+            log::info!(
+                "Publish succeeded, but package was not published because it was run in dry-run mode"
+            );
+
+            return Ok(());
+        }
+
+        let namespace = self
+            .namespace
+            .as_deref()
+            .or_else(|| package.name.split('/').next())
+            .unwrap_or("")
+            .to_string();
+        if username != namespace {
+            return Err(anyhow::anyhow!("trying to publish package under the namespace {namespace:?}, but logged in as user {username:?}"));
+        }
```

Comment:
Won't this error out when someone is publishing to a namespace (e.g. when `ciuser` tries publishing to `wasmer`)?

Reply:
Hm, yeah, we should have a query `can_publish_package` - I just wanted to avoid that the user has to wait for uploading and then at the end it errors out with "cannot publish package, you don't have rights".

Reply:
It's implemented now, but the functionality of --namespace is still a bit lacking since it doesn't allow you to change the namespace when uploading.

Reply:
This will be resolved with the new `wapm-toml` version.

Reply:
I removed this check now and use `--package-name` instead